app/src/main/java/ca/bc/gov/bchealth/ui/custom/DatePickerExtension.kt

```diff
@@ -0,0 +1,70 @@
+package ca.bc.gov.bchealth.ui.custom
+
+import androidx.core.widget.doOnTextChanged
+import androidx.fragment.app.Fragment
+import ca.bc.gov.bchealth.R
+import ca.bc.gov.bchealth.utils.adjustOffset
+import com.google.android.material.datepicker.MaterialDatePicker
+import com.google.android.material.textfield.TextInputLayout
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+/*
+* @author amit_metri on 13,January,2022
+*/
+
+val simpleDateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.CANADA)
+
+fun Fragment.setUpDatePickerUi(tipDatePicker: TextInputLayout, tag: String) {
+
+    val dateOfBirthPicker =
+        MaterialDatePicker.Builder.datePicker()
+            .setTitleText(getString(R.string.select_date))
+            .setSelection(MaterialDatePicker.todayInUtcMilliseconds())
+            .build()
+    tipDatePicker.editText?.setOnClickListener {
+        dateOfBirthPicker.show(parentFragmentManager, tag)
+    }
+    tipDatePicker.setEndIconOnClickListener {
+        dateOfBirthPicker.show(parentFragmentManager, tag)
+    }
+    dateOfBirthPicker.addOnPositiveButtonClickListener {
+        tipDatePicker.editText?.setText(simpleDateFormat.format(it.adjustOffset()))
+    }
+}
+
+fun Fragment.validateDatePickerData(
+    tipDatePicker: TextInputLayout,
+    errorMessage: String
```

Comment:
`errorMessage` here is only if it is missing right? Seems like a badly named variable in that case. Also, why is this error message injected as a param but the other one is a lookup.

Reply:
This will go away with the gradual refactoring